examples/custom-node/src/primitives/extended_op_tx_envelope.rs

```diff
@@ -0,0 +1,303 @@
+            Self::BuiltIn(tx) => tx.blob_versioned_hashes(),
+            Self::Other(tx) => tx.blob_versioned_hashes(),
+        }
+    }
+    fn authorization_list(&self) -> Option<&[SignedAuthorization]> {
+        match self {
+            Self::BuiltIn(tx) => tx.authorization_list(),
+            Self::Other(tx) => tx.authorization_list(),
+        }
+    }
+}
+
+impl<T> SignedTransaction for ExtendedOpTxEnvelope<T>
+where
+    T: SignedTransaction,
+{
+    fn recover_signer(&self) -> Result<Address, RecoveryError> {
+        match self {
+            Self::BuiltIn(tx) => tx.recover_signer(),
+            Self::Other(tx) => tx.recover_signer(),
+        }
+    }
+
+    fn recover_signer_unchecked(&self) -> Result<Address, RecoveryError> {
+        match self {
+            Self::BuiltIn(tx) => tx.recover_signer_unchecked(),
+            Self::Other(tx) => tx.recover_signer_unchecked(),
+        }
+    }
+
+    fn recover_signer_unchecked_with_buf(
+        &self,
+        buf: &mut Vec<u8>,
+    ) -> Result<Address, RecoveryError> {
+        match self {
+            Self::BuiltIn(tx) => tx.recover_signer_unchecked_with_buf(buf),
+            Self::Other(tx) => tx.recover_signer_unchecked_with_buf(buf),
+        }
+    }
+
+    fn tx_hash(&self) -> &TxHash {
+        match self {
+            Self::BuiltIn(tx) => tx.tx_hash(),
+            Self::Other(tx) => tx.tx_hash(),
+        }
+    }
+
+    fn signature(&self) -> &PrimitiveSignature {
+        match self {
+            Self::BuiltIn(tx) => tx.signature(),
+            Self::Other(tx) => tx.signature(),
+        }
+    }
+}
+
+impl<T> Typed2718 for ExtendedOpTxEnvelope<T>
+where
+    T: Typed2718,
+{
+    fn ty(&self) -> u8 {
+        match self {
+            Self::BuiltIn(tx) => tx.ty(),
+            Self::Other(tx) => tx.transaction.tx_type() as u8,
+        }
+    }
+}
+
+impl<T> Decodable2718 for ExtendedOpTxEnvelope<T>
+where
+    T: Decodable2718,
+{
+    fn typed_decode(ty: u8, buf: &mut &[u8]) -> Eip2718Result<Self> {
+        OpTxEnvelope::typed_decode(ty, buf)
+            .map(Self::BuiltIn)
+            .or_else(|_| OpTransactionSigned::typed_decode(ty, buf).map(Self::Other))
+    }
+
+    fn fallback_decode(buf: &mut &[u8]) -> Eip2718Result<Self> {
+        OpTxEnvelope::fallback_decode(buf)
+            .map(Self::BuiltIn)
+            .or_else(|_| OpTransactionSigned::fallback_decode(buf).map(Self::Other))
+    }
+}
+
+impl<T> Decodable for ExtendedOpTxEnvelope<T>
+where
+    T: Decodable,
+{
+    fn encode_2718(&self, out: &mut dyn BufMut) {
+        match self {
+            Self::BuiltIn(tx) => tx.encode_2718(out),
+            Self::Other(tx) => tx.encode_2718(out),
+        }
+    }
+
+    fn encode_2718_len(&self) -> usize {
+        match self {
+            Self::BuiltIn(tx) => tx.encode_2718_len(),
+            Self::Other(tx) => tx.encode_2718_len(),
+        }
+    }
+}
+
+impl<T> Encodable for ExtendedOpTxEnvelope<T>
+where
+    T: Encodable,
+{
```

Comment:
the `Encodable` and `Decodable` impls are swapped

Reply:
done